code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -25,33 +25,122 @@
+	QDEL_NULL(goosevomit)
+	return ..()
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/examine(user)
 	. = ..()
-	if(resting)
-		wander = FALSE
-	else
-		wander = TRUE
-	update_icon()
+	to_chat(user, "<span class='notice'>Somehow, it still looks hungry.</span>")
 
-/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/attacked_by(obj/item/O, mob/user)
 	. = ..()
-	if(prob(5))
-		Retaliate()
+	feed(O)
 
-/mob/living/simple_animal/hostile/retaliate/goose/proc/update_icon()
-	if(stat != DEAD)
-		icon_state = resting ? icon_resting : icon_living
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/feed(obj/item/O)
+	. = ..()
+	if(!istype(O, /obj/item/reagent_containers/food))
+		return
+	var/obj/item/reagent_containers/food/tasty = O
+	if (tasty.foodtype & GROSS)
+		visible_message("<span class='notice'>[src] hungrily gobbles up \the [tasty]!</span>")
+		tasty.forceMove(src)
+		playsound(src,'sound/items/eatfood.ogg', 70, 1)
+		vomitCoefficient ++
+		vomitTimeBonus += 5
+	else
+		visible_message("<span class='notice'>[src] refuses to eat \the [tasty].</span>")
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit()
+	var/turf/T = get_turf(src)
+	var/obj/item/reagent_containers/food/consumed = locate() in contents //Barf out a single food item from our guts
+	if (prob(50) && consumed)
+		barf_food(consumed)
 	else
-		icon_state = icon_dead
+		playsound(get_turf(src), 'sound/effects/splat.ogg', 50, 1) //yes getting the turf twice is necessary fuck off
+		T.add_vomit_floor(src)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/barf_food(var/atom/A)
+	if(istype(A, /obj/item/reagent_containers/food))
+		return
+	var/turf/T1 = get_turf(src) //its not technically a one-letter var, checkmate ninjanom
+	var/obj/item/reagent_containers/food/consumed = A
+	consumed.forceMove(T1)
+	var/destination = get_edge_target_turf(T1, pick(GLOB.alldirs)) //Pick a random direction to toss them in
+	consumed.throw_at(destination, 1, 2) //Thow the food at a random tile 1 spot away
+	var/turf/T2 = get_turf(consumed)
+	T2.add_vomit_floor(src)
+	playsound(get_turf(consumed), 'sound/effects/splat.ogg', 50, 1) //yes getting the turf twice is necessary fuck off
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit_start(duration)
+	flick("vomit_start",src)
+	vomiting = TRUE
+	icon_state = "vomit"
+	vomit()
+	addtimer(CALLBACK(src, .proc/vomit_end), duration)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit_end()
+	for (var/obj/item/consumed in contents) //Get rid of any food left in the poor thing
+		barf_food(consumed)
+	flick("vomit_end",src)
+	vomiting = FALSE
+	icon_state = initial(icon_state)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Moved(oldLoc, dir)
+	. = ..()
+	if(vomiting)
+		vomit() // its supposed to keep vomiting if you move
+		return
+	for (var/tasty in get_turf(src))
```

Comment:
you should put a maximum limit on what it will eat otherwise it could really mess up a chefs day

Reply:
shit good idea

Reply:
@optimumtact I can't really see it messing up a chef's day, as it will only eat gross food, so shit like rats and trash.
Furthermore, even if it does eat the chef's private reserve of gross food, it'll just barf it back up before getting too far away (as barf-chance is rolled on tile moved)

Reply:
CHECK_TICK at least is merited though

Reply:
okay, but it's still going through everything on the turf to achieve this, which is suboptimal.

Reply:
you should at least do `for(var/foodpath blah/tasty)`

Reply:
done